### PR TITLE
Ugact 24 support es6

### PR DESCRIPTION
**Pull request by KamilChoroba**
/cc @AutoScout24/web-experience

## Description
To use newer libraries like https://roughnotation.com/ we need to be able to support ES6 modules.
This requires and upgrade of the showcar-ui bundler.

## Risks
- [medium] Describe possible risk here
  * Replaced showcar-gulp bundler with native gulp
  * Support ES6 libraries

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -1,70 +1,121 @@
...
+const eslint = require('gulp-eslint');
+const webpack = require('webpack-stream');
+const UglifyJsPlugin = require('uglifyjs-webpack-plugin');
+
+const destination = 'dist';
+
+const generateWebpackOptions = (output) => {
+    const filename = output;
+    return {
+        mode: 'development',
```
Comment (yifan9916) [inline gulpfile.js:11]: ideally the environment should be set with `NODE_ENV` so this line would be unnecessary. could you clarify?

File: gulpfile.js [hunk 2]
```diff
@@ -1,70 +1,121 @@
...
-gulp.task('serve', scgulp.serve({
-    dir: 'dist'
-}));
+const eslint = require('gulp-eslint');
+const webpack = require('webpack-stream');
+const UglifyJsPlugin = require('uglifyjs-webpack-plugin');
+
+const destination = 'dist';
+
+const generateWebpackOptions = (output) => {
```
Comment (yifan9916) [inline gulpfile.js:8]: is it difficult to use a webpack.config file instead?

File: gulpfile.js [hunk 3]
```diff
@@ -1,70 +1,121 @@
...
+                options: {
+                    presets: [['@babel/preset-env', { modules: false }]],
+                    cacheDirectory: '.tmp/js',
+                },
+            }],
+        },
+        plugins: [],
+        optimization: {
+            minimizer: [
+                new UglifyJsPlugin({
```
[the remark this answers is not in the document]
Comment (KamilChoroba) [reply gulpfile.js:29]: Changed

File: gulpfile.js (hunk 2 above)
Comment (KamilChoroba) [reply gulpfile.js:8]: True, it's a better approach. Done

File: gulpfile.js (hunk 1 above)
Comment (KamilChoroba) [reply gulpfile.js]: Fixed

**Pull request merged**